### DISCUSSION TITLE
Rain simulation

### DISCUSSION THREAD

**Discussion by mariuszhermansdorfer** [InteractiveComputerGraphics/SPlisHSPlasH#256] [General]
I’d like to ask whether it’s possible to use this library for surface water flow simulations similar to how rainwater would flow on a terrain?

**Comment by janbender** [comment]
Yes, that would be possible.

**Reply by mariuszhermansdorfer** [reply]
Thanks @janbender. Could you please point me in the right direction? A relevant function or a code snippet doing something similar would be great!

**Comment by janbender** [comment]
I would need some more info. There are several examples which show the fluid flowing on top of a boundary geometry. Do you want to emit the rain drops somehow?

**Reply by mariuszhermansdorfer** [reply]
I'd have a triangulated mesh representing a terrain and would like to simulate a uniform rain event over the entire domain. Each cell would receive the same amount of water over a specified period. Ideally, I'd like to vary the rain intensity over time (start with less, increase the amount at peak, and taper off towards the end).

**Comment by janbender** [comment]
I think the best way to perform such a simulation is to implement a new rain emitter with the desired properties. Take a look at the already implemented emitter and modify it.
